src/components/tokenInput/tokenInput.js

```diff
@@ -1,16 +1,42 @@
-import './tokenInput.css';
+import { useState } from "react";
+import "./tokenInput.css";
 
-function TokenInput(props) {
-    const {onSubmit} = props
-    return (
-        <div>
-            <h1>Enter Slack Token</h1>
-            <form onSubmit={onSubmit}>
-                <input type="password" />
-                <input type="submit" />
-            </form>
-        </div>
-    );
+function TokenInput({ login }) {
+  const [token, setToken] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [tokenValid, setTokenValid] = useState(null);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setLoading(true);
+    const valid = await login(token);
+    if (!valid) {
+      setTokenValid(false);
+      setLoading(false);
```

Comment:
Add simple alert back in
```suggestion
      setLoading(false);
      alert(`Invalid token: ${token}`);
```